arcaflow_plugin_fio/fio_schema.py

```diff
@@ -56,110 +62,443 @@ def is_sync(self) -> bool:
         return self.value in self._sync_io_engines
 
 
+duration_pattern_string = r"[1-9][0-9]*(?:d|h|m|s|ms|us)?"
+duration_pattern = re.compile(rf"^{duration_pattern_string}$")
+duration_range_pattern = re.compile(
+    rf"^{duration_pattern_string}(?:-{duration_pattern_string})?$"
+)
+
+size_pattern_string = (
+    r"0x[0-9a-fA-F]+|"
+    r"\d+(?:k|kb|ki|kib|m|mb|mi|mib|g|gb|gi|gib|t|tb|ti|tib|p|pb|pi|pib)?"
+)
+size_pattern = re.compile(
+    rf"^(?:{size_pattern_string})(?:,(?:{size_pattern_string})){{0,2}}$",
```

Comment:
Does this really allow just 1 - 3 size patterns? I must misunderstand ... I'd thought the comma lists were "per-thread" values, which were more or less unbounded?

Reply:
The three values are ["read, write, and trim"](https://github.com/arcalot/arcaflow-plugin-fio/blob/422bb00356e3c1c3fe1d770a59240f9055c698f5/arcaflow_plugin_fio/fio_schema.py#L310-L313), and, if you omit the second or third then the previous value is reused, which is why zero, one, or two values are accepted.